content/doc/book/system-administration/diagnosing-errors.adoc

```diff
@@ -0,0 +1,62 @@
+---
+layout: section
+---
+
+ifdef::backend-html5[]
+ifndef::env-github[:imagesdir: ../../resources]
+:notitle:
+:description:
+:author:
+:email: jenkinsci-users@googlegroups.com
+:sectanchors:
+:toc: left
+endif::[]
+
+= Diagnosing Errors
+
+If your Jenkins started choking with `OutOfMemoryError`, there are four possibilities.
+
+- Your Jenkins is growing in data size, requiring a bigger heap space. In this case you just want to give it a bigger heap.
+- Your Jenkins is temporarily processing a large amount of data (like test reports), requring a bigger head room in memory. In this case you just want to give it a bigger heap.
+- Your Jenkins is leaking memory, in which case we need to fix that.
+- The Operating System kernel is running out of virtual memory.
+
+Which category your `OutOfMemoryError` falls into is not always obvious, but here are a few useful techniques to diagnose the problem.
+
+- Use https://visualvm.github.io/[VisualVM], attach to the running instance, and observe the memory usage. Does the memory max out while loading Jenkins? If so, it probably just needs a bigger memory space. Or is it slowing creeping up? If so, maybe it is a memory leak.
```

Comment:
The original visualvm link was a broken link shown in this picture. https://visualvm.github.io/ appears to be the newer visualvm link.

![image](https://user-images.githubusercontent.com/5103752/96719242-c3f6e500-136e-11eb-8739-4543a9fe9dee.png)